### PR TITLE
feat(protocol): add withdraw eth function to proverset

### PR DESCRIPTION
**Pull request by RogerLamTd**
ref #17784

### CODE REVIEW

File: packages/protocol/contracts/team/proving/ProverSet.sol [hunk 1]
```diff
@@ -77,6 +77,16 @@ contract ProverSet is EssentialContract, IERC1271 {
         IERC20(tkoToken()).transfer(admin, _amount);
     }
 
+    /// @notice Withdraws ETH back to the owner address.
+    function withdraw(uint256 _amount) external onlyOwner {
```
Comment (dantaik) [inline packages/protocol/contracts/team/proving/ProverSet.sol:81]:
```suggestion
    function withdrawEther(uint256 _amount) external onlyOwner {
```

File: packages/protocol/contracts/team/proving/ProverSet.sol [hunk 2]
```diff
@@ -77,6 +77,16 @@ contract ProverSet is EssentialContract, IERC1271 {
...
+        uint256 balance = address(this).balance;
+        require(balance - _amount > 0, "Amount to withdraw > balance of smart contract");
```
Comment (dantaik) [inline packages/protocol/contracts/team/proving/ProverSet.sol:83]: `if (balance < amount) revert SOME_ERROR_HERE`

File: packages/protocol/contracts/team/proving/ProverSet.sol [hunk 3]
```diff
@@ -77,6 +77,16 @@ contract ProverSet is EssentialContract, IERC1271 {
...
+
+        address _owner = owner();
+        (bool sent,) = _owner.call{ value: _amount }("");
+        require(sent, "Failed to withdraw ETH");
```
Comment (dantaik) [inline packages/protocol/contracts/team/proving/ProverSet.sol:87]: use `sendEtherAndVerify` in LibAddress

File: packages/protocol/contracts/team/proving/ProverSet.sol [hunk 4]
```diff
@@ -77,6 +79,15 @@ contract ProverSet is EssentialContract, IERC1271 {
         IERC20(tkoToken()).transfer(admin, _amount);
     }
 
+    /// @notice Withdraws ETH back to the owner address.
+    function withdrawEther(uint256 _amount) external onlyOwner {
```
Comment (dantaik) [inline packages/protocol/contracts/team/proving/ProverSet.sol:83]:
```suggestion
    function withdrawEtherToAdmin(uint256 _amount) external onlyAuthorized {
```

File: packages/protocol/contracts/team/proving/ProverSet.sol [hunk 5]
```diff
@@ -77,6 +79,15 @@ contract ProverSet is EssentialContract, IERC1271 {
...
+        uint256 balance = address(this).balance;
+        if (balance < _amount) revert AMOUNT_GT_BALANCE();
+
+        address _owner = owner();
+        LibAddress.sendEtherAndVerify(_owner, _amount);
```
Comment (dantaik) [inline packages/protocol/contracts/team/proving/ProverSet.sol:88]:
```suggestion
        LibAddress.sendEtherAndVerify(admin, _amount);
```

**Review by dantaik — APPROVED** [review]

**Review by davidtaikocha — APPROVED** [review]

**Pull request merged**